pydantic/json.py

```diff
@@ -20,7 +20,32 @@
 from .networks import NameEmail
 from .types import SecretBytes, SecretStr
 
-__all__ = 'pydantic_encoder', 'custom_pydantic_encoder', 'timedelta_isoformat'
+__all__ = (
+    'BuiltinTypeWrapper',
+    'pydantic_encoder',
+    'custom_pydantic_encoder',
+    'timedelta_isoformat',
+    'wrap_builtin_types',
+)
+
+
+class BuiltinTypeWrapper:
+    def __init__(self, obj: Any) -> None:
+        self.obj = obj
+
+
+def wrap_builtin_types(obj: Any, types: AbstractSet[Type[Any]]) -> Any:
+    if obj.__class__ is dict:
+        obj = {k: wrap_builtin_types(v, types) for k, v in obj.items()}
+    elif obj.__class__ is list:
+        obj = [wrap_builtin_types(v, types) for v in obj]
+    elif obj.__class__ is tuple:
+        obj = tuple(wrap_builtin_types(v, types) for v in obj)
+
+    if obj.__class__ in types:
```

Comment:
What happens to subclass of these types? Is there a reason not to use `isinstance`?

Reply:
Nope, it was a mistake from my side. Fixed.